lib/widgets/compose_box.dart

```diff
@@ -1523,14 +1523,27 @@ class _ComposeBoxState extends State<ComposeBox> with PerAccountStoreAwareStateM
 
   @override
   void onNewStore() {
+    final newStore = PerAccountStoreWidget.of(context);
+
+    if (_controller == null) {
+      _setNewControllerForNarrow(newStore);
+      return;
+    }
+
+    switch (_controller) {
+      case StreamComposeBoxController():
+        (_controller as StreamComposeBoxController).topic.store = newStore;
+      case FixedDestinationComposeBoxController():
+        break; // no reference to the store that needs updating
+      case null:
+        assert(false);
+    }
+  }
+
+  void _setNewControllerForNarrow(PerAccountStore store) {
```

Comment:
nit: drop "for narrow"? Seems like it's just the new controller that's appropriate for this compose box.